commit d7873cb9abb5d8b4b9f7f5749af06e4e03798733
Author: Jouni Malinen <jouni.malinen@atheros.com>
Date:   Fri Mar 20 15:53:16 2009 +0200

    mac80211: Fix memleak in nl80211 authentication on deinit
    
    This file was forgotten from the quilt patch that added MLME
    primitives, so the kfree on interface removal is missing. Fix this
    potential memleak by freeing the temporary Authentication frame IEs
    from SME when the interface is being removed.
    
    Signed-off-by: Jouni Malinen <jouni.malinen@atheros.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/iface.c b/net/mac80211/iface.c
index f9f27b9cadbe..6b56dc2208e7 100644
--- a/net/mac80211/iface.c
+++ b/net/mac80211/iface.c
@@ -660,6 +660,7 @@ static void ieee80211_teardown_sdata(struct net_device *dev)
 		kfree(sdata->u.mgd.ie_reassocreq);
 		kfree(sdata->u.mgd.ie_deauth);
 		kfree(sdata->u.mgd.ie_disassoc);
+		kfree(sdata->u.mgd.sme_auth_ie);
 		break;
 	case NL80211_IFTYPE_WDS:
 	case NL80211_IFTYPE_AP_VLAN: